src/mmrelay/plugin_loader.py

```
@@ -870,323 +927,634 @@ def _check_auto_install_enabled(config):
 
     Returns:
-        bool: `True` if the repository was successfully cloned or updated, `False` otherwise.
+        ValidationResult: NamedTuple with fields:
+            - is_valid (bool): `True` if inputs are valid, `False` otherwise.
+            - repo_url (str|None): The normalized repository URL on success, `None` on failure.
+            - ref_type (str|None): One of "tag", "branch", or "commit" on success, `None` on failure.
+            - ref_value (str|None): The validated ref value on success, `None` on failure.
+            - repo_name (str|None): Derived repository name (basename without extension) on success, `None` on failure.
+
+    Notes:
+        - Commit `value` must be 7–40 hexadecimal characters.
```

Ruff (0.14.4)

Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)